### PR TITLE
Omit some lookups in gateway

### PR DESCRIPTION
**Pull request by butonic**
The CS3 gateway can directly to itself when it wants to talk to the registries running in the same reva runtime.

Requires a reva bump that is part of this PR to pull in https://github.com/cs3org/reva/pull/4744

part of https://github.com/owncloud/ocis/issues/8589

### CODE REVIEW

File: services/gateway/pkg/revaconfig/config.go [hunk 1]
```diff
@@ -12,6 +12,17 @@ import (
 
 // GatewayConfigFromStruct will adapt an oCIS config struct into a reva mapstructure to start a reva service.
 func GatewayConfigFromStruct(cfg *config.Config, logger log.Logger) map[string]interface{} {
+	self := cfg.GRPC.Addr
```
Comment (kobergj) [inline services/gateway/pkg/revaconfig/config.go:15]: Buh! What is `self` supposed to mean? Does this code think it is java?

File: services/storage-system/pkg/revaconfig/config.go [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
 import (
+	"strings"
+
 	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
 	"github.com/owncloud/ocis/v2/services/storage-system/pkg/config"
 )
 
 // StorageSystemFromStruct will adapt an oCIS config struct into a reva mapstructure to start a reva service.
 func StorageSystemFromStruct(cfg *config.Config) map[string]interface{} {
+	self := cfg.GRPC.Addr
```
Comment (kobergj) [inline services/storage-system/pkg/revaconfig/config.go:12]: Same here. Please don't write java :cry: Call it `grpcAddr` (or something else that describes what it is)

**Review by kobergj — CHANGES_REQUESTED** [review]
Protest with naming. Code looks good though :+1:

**Comment by butonic** [comment]
@kobergj mea culpa

**Review by kobergj — APPROVED** [review]
Awesome :+1:

**Pull request merged**